py/torch_tensorrt/_Input.py

```diff
@@ -364,9 +364,13 @@ def example_tensor(
                     )
 
                 if isinstance(self.shape, dict):
-                    return torch.rand(self.shape[optimization_profile_field]).to(
-                        dtype=self.dtype.to(torch.dtype, use_default=True)
```

Comment:
If dtype is torch.bool in input_spec, torch.rand() returns random numbers on [0,1) and casting it to bool value is True